lib/route.js

```diff
@@ -137,11 +141,19 @@ function buildRouting (options) {
         if (supportedMethods.indexOf(method) === -1) {
```

Comment:
Can you include all the logic of validating `method` and `schema.body` inside the extracted function. It should be something similar to `validateBodyLimitOption`. You may name it as `validateMethodAndSchemaBodyOption`.